cmd/infra/aws/destroy_iam.go

```diff
@@ -50,12 +52,27 @@ func NewDestroyIAMCommand() *cobra.Command {
 			<-sigs
 			cancel()
 		}()
-		return opts.DestroyIAM(ctx)
+		if err := opts.DestroyIAM(ctx); err != nil {
+			return err
+		}
+		log.Info("Successfully destroyed IAM infra")
+		return nil
 	}
 
 	return cmd
 }
 
+func (o *DestroyIAMOptions) Run(ctx context.Context) error {
+	return wait.PollUntil(5*time.Second, func() (bool, error) {
+		err := o.DestroyIAM(ctx)
+		if err != nil {
+			log.Error(err, "error destroying infra")
```

Comment:
Can we make this a warning? It's confusing to see an error getting logged but the command continues.

Reply:
Yeah, will wait for e2e to finish